Automattic/wp-calypso
# Revert "Revert "Onboarding: show the draft post modal

#### Changes proposed in this Pull Request

* Re-deploy [show the draft post modal](https://github.com/Automattic/wp-calypso/pull/56750). As there is an issue https://github.com/Automattic/wp-calypso/issues/56930, we reverted the changes in this PR: https://github.com/Automattic/wp-calypso/pull/56953
* Replace `useI18n` with `__` from `@wordpress/i18n`. If the Gutenberg is deactivate, using `useI18n` leads to `wp-nux` doesn't load so that the following action becomes `undefined` and the user cannot close page patterns modal
  ```js
  const { setUsedPageOrPatternsModal } = useDispatch( 'automattic/wpcom-welcome-guide' );
   ```

#### Testing instructions



##### Test draft post modal

* Sandbox your API and ensure your siteSlug domain point to your sandbox
* Open another terminal and run cd apps/editing-toolkit && yarn dev --sync
* Go to /start/setup-site?siteSlug=arthurtestonboarding.wordpress.com&flags=signup/hero-flow
* Select write intent and you'll redirect to post editor
* Verify write your post modal show
* Verify the following behaviors one by one (note that when you finish one, you need to repeat to choose the intent)
  * Click [X] to verify the modal is closed and stay in the editor
  * Click [Start writing] to verify the modal closed and stay in the editor
  * Click [I'm not ready] to verify you go to My Home
* Refresh the page and verify the write your post modal not show but welcome tour shows

##### Test Add page template modal on AT site

* Deactivate Gutenberg, activate WordPress.com Editing Toolkit plugin
* Visit "My Site > Pages", select "Add New" - (wp-admin/post-new.php?post_type=page)
* Either either one of pre-defined layouts, or "Blank Page"
* Select 'x' to edit creating new page template.



Related to https://github.com/Automattic/wp-calypso/issues/56930, https://github.com/Automattic/wp-calypso/pull/56953, https://github.com/Automattic/wp-calypso/pull/56750
